tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,178 @@ Take care not to keep any important state in your renderer - animation tickers
+
+#### Create a Widget Renderer
+
+The widget renderer is how we tell the fyne application what to draw and how to lay it out.
+
+The widget renderer has it's own state and functions (see ```fyne.WidgetRenderer``` interface) but this should only be concerned with drawing (rendering) the widget.
+
+The renderer is responsible for:
+
+1) The objects that are to be drawn. These are returned by the ```Objects() []fyne.CanvasObject``` function. Notice that you can return any object as long as it implements the ```fyne.CanvasObject``` interface.
+2) Each object returned by the Objects() function will need to be created by the widget renderer and initialised before returning.
+3) The fyne application will then call the renderers Layout() function. This is where the size and position of each object is detirmined.
+
+Our example renderer could be stored in the following struct:
+
+```
+type myWidgetRenderer struct {
+	minSize fyne.Size // The size of the widget
+	rect    *canvas.Rectangle
+	text    *canvas.Text
+}
+```
+
+As you can see it is about canvas objects that the fyne application will render. A lower case name is used as the renderer code is private and only created by the Widget. 
+
+The constructor is the place to pass in properties like minimum size, text values and other data.
+
+```go
+func newMyWidgetRenderer(size fyne.Size, text string) *myWidgetRenderer {
+	return &myWidgetRenderer{
+		minSize: size,  // For the MinSize() function
+		rect:    canvas.NewRectangle(theme.BackgroundColor()),
+		text:    canvas.NewText(text, theme.ForegroundColor()),
+	}
+}
+```
+
+Notice that the size and position of the canvas objects are not defined here, only colours, text and other properties. It might be appropriate to pass in a reference to the widget instead of lots of different parameters. This is left to your design requirements.
+
+The Objects() function can now return the canvas objects as a list:
+
+```go
+func (r *myWidgetRenderer) Objects() []fyne.CanvasObject {
+    // The order is critical, rect is drawn first then text
+	return []fyne.CanvasObject{r.rect, r.text}
+}
+```
+
+Each object returned here will need to be sized and positioned in the Layout() function:
+
+```go
+func (r *myWidgetRenderer) Layout(s fyne.Size) {
+	// Measure the size of the text so we can Center it
+	si := fyne.MeasureText(r.text.Text, r.text.TextSize, r.text.TextStyle)
+	r.text.Move(fyne.Position{X: (s.Width - si.Width) / 2, Y: (s.Height - si.Height) / 2})
+	// Resize the background
+	r.rect.Resize(s)
+}
+```
+
+We finish with the remaining functions of the ```fyne.WidgetRenderer``` interface
+
+```go
+func (r *myWidgetRenderer) Destroy() {} // Called when the renderer is destroyed
+func (r *myWidgetRenderer) Refresh() {} // This does not seem to be used!
+func (r *myWidgetRenderer) MinSize() fyne.Size {
+	return r.minSize // Return the minimum size of the widget
```

Comment:
You should use measureText, as above. We must always return the true minimum size of this widget so that the UI can adapt appropriately to all screens.